### DISCUSSION TITLE
signclient having the issue with aplha wallet when pop up appears its says all event must be approved

### DISCUSSION THREAD

**Discussion by arbaz-meo** [WalletConnect/walletconnect-monorepo#3347] [Developer Support] [answered]
aplha wallet having the issue with wallect connect signclient . before moving to 2.9.2 its was not present

**Comment by arbaz-meo** [comment]
@0xAsimetriq can you look this issue

**Reply by arbaz-meo** [reply]
@0xAsimetriq  can you tell me about this why its happening

**Comment by glitch-txs** [comment]
Hi, could you share a minimal reproducible example to test this? 
 Do you experience this on labs as well? https://lab.web3modal.com/

**Reply by arbaz-meo** [reply]
i m using signclient simply so i m not using any kind of model , i m generating QR code by getting uri from signclient and its working with metamask wallet and trust wallet   but its showing me error on alpha wallet  

![MicrosoftTeams-image (3)](https://github.com/WalletConnect/walletc
![MicrosoftTeams-image (4)](https://github.com/WalletConnect/walletconnect-monorepo/assets/74611973/c635ffa0-27f7-4c6b-adfb-8bcefe64a5ca)
![MicrosoftTeams-image (5)](https://github.com/WalletConnect/walletconnect-monorepo/assets/74611973/837d5f9e-15b3-4a79-91ef-c1ff8ce87f9e)

onnect-monorepo/assets/74611973/54ab200f-f578-4ef2-ab27-4a7f911c45c7)


i attach these images to better understanding for issue

**Reply by arbaz-meo** [reply]
@glitch-txs before this issue was not exist

**Reply by arbaz-meo** [reply]
@finessevanes  can you also look this issue please

**Reply by glitch-txs** [reply]
can you share your configuration?

**Reply by arbaz-meo** [reply]
```
const{uri,approval}=  await Client.connect({
       requiredNamespaces: {
         eip155: {
           methods: ['eth_sign','eth_sendTransaction'],
           chains: ['eip155:80001'],
           events: ["connect",'accountsChanged']
         }
       }
```

@glitch-txs  here you go

**Reply by glitch-txs** [reply]
does the issue persists if you add `chains: ['eip155:1', 'eip155:80001']` ?

**Reply by arbaz-meo** [reply]
```
Client = await SignClient.init({ projectId ,  metadata: {
    name: 'MyWhoosh',
    description: 'MYwhoosh Dapp',
    url: 'https://walletconnect.com/', // Do we have this one
    icons: ['https://www.mywhoosh.com/wp-content/uploads/2022/10/my-whoosh-logo-white-v3.png'], // Looks like we need to deploy our splash somewhere
  
  },
},
);
```

**Reply by arbaz-meo** [reply]
@glitch-txs let me check

**Reply by arbaz-meo** [reply]
@glitch-txs yes issue still exist on main chain

**Reply by glitch-txs** [reply]
iOS or android? could you share a very minimal example that reproduces this issue? working on my side for android

**Reply by arbaz-meo** [reply]
@glitch-txs for me its having the issue on alpha wallet version 3.60.8 and sign client version 2.9.2  . so for me its also  working on trust wallet and metamask 

```

const  {SignClient} = require ('@walletconnect/sign-client');
const uriMap = new Map();
const connectedMap = new Map();
const projectId = process.env.ProjectId;

let Client = {};

const intilizeSignClient = async () => {
 Client = await SignClient.init({ projectId ,  metadata: {
    name: 'MyWhoosh',
    description: 'MYwhoosh Dapp',
    url: 'https://walletconnect.com/', // Do we have this one
    icons: ['https://www.mywhoosh.com/wp-content/uploads/2022/10/my-whoosh-logo-white-v3.png'], // Looks like we need to deploy our splash somewhere
  
  },


},


);


   
   
} 



/*  
  we are paring session topic  every time for every user on behalf of DeviceID 
*/

async function connectWallet(deviceId){
  try {
   // intilizeSignClient();
  
    const{uri,approval}=  await Client.connect({
       requiredNamespaces: {
         eip155: {
           methods: ['eth_sign','eth_sendTransaction'],
           chains: ['eip155:1'],
           events: ["connect",'accountsChanged']
         }
       },
     
    
      
     })
     console.log(uri);
     uriMap.set(deviceId,uri);
     connectedMap.set(deviceId,approval());
   

  } catch(e){
    console.log(e.message);
  }

  


}

const getClient = () => {
  return Client
}

module.exports.connectWallet=connectWallet;
module.exports.uriMap=uriMap;
module.exports.connectedMap=connectedMap;
module.exports.signClient = getClient;
module.exports.intilizeSignClient = intilizeSignClient ;



```
@glitch-txs  here is the full code

**Reply by arbaz-meo** [reply]
its android

**Reply by glitch-txs** [reply]
can you try replacing "eth_sign" with "personal_sign"?

**Reply by arbaz-meo** [reply]
@glitch-txs sure

**Reply by arbaz-meo** [reply]
@glitch-txs again its work with metamask while says on alpha wallet **session has been terminated so please switch back to Dapp and start new**

**Reply by arbaz-meo** [reply]
@glitch-txs 1 month ago its was working fine for me with same code  on trust wallet

**Reply by glitch-txs** [reply]
mmm it's weird because it's working on my side, have you tried reinstalling the app? could you also check on labs? it's the same config

**Reply by arbaz-meo** [reply]
@glitch-txs  i test mydapp on wallect connect vercel app and i sucessfully connect the wallet but still not working on aplha wallet .
![Capture](https://github.com/WalletConnect/walletconnect-monorepo/assets/74611973/1c3d2101-8645-4a8f-95d5-9fceea37ade9)

**Reply by glitch-txs** [reply]
just checked on my version, Alpha Wallet 3.65, you might need to update it :)

**Reply by arbaz-meo** [reply]
@glitch-txs  i just test on i phone its work but still not work on android with 3.65 too

**Comment by arbaz-meo** [comment, marked as answer]
@glitch-txs  issue is fixed the issue related was event . i just remove the event connect and replace with chainchanged   `['chainChanged', 'accountsChanged'] ` . its should not be happen but its working
